plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentPluginImpl.kt

```diff
@@ -130,6 +131,10 @@ class LocationComponentPluginImpl : LocationComponentPlugin, LocationConsumer,
   private fun activateLocationComponent() {
     if (internalSettings.enabled) {
       delegateProvider.getStyle { style ->
+        if (locationComponentActivatedStyle == style && isLocationComponentActivated) {
```

Comment:
why do we need to compare the two `StyleManagerInterface`? could we use the `LocationPuckManager#isLayerInitialised` to check if the style is currently under use?

Reply:
I thought about it but I decided that my way is safer in case we trigger that lambda with some new style while old one has changed.
However I do not have all the context here, your way works better, so will do like you proposed.